### PR TITLE
feat(ctb): remove blockedSystemAddresses

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Removes the blockedSystemAddresses in favor of a more flexible approach that also removes most of the need for the initializer. We will be following this up with a second PR that removes the initializer entirely.

**Tests**
No tests required, same logic as before is being tested.

### CODE REVIEW

File: packages/contracts-bedrock/tasks/genesis-l2.ts [hunk 1]
```diff
@@ -248,9 +248,7 @@ task('genesis-l2', 'create a genesis config')
         _owner: deployConfig.l2CrossDomainMessengerOwner,
         xDomainMsgSender: '0x000000000000000000000000000000000000dEaD',
         msgNonce: 0,
-        otherMessenger: ProxyL1CrossDomainMessenger.address,
-        // TODO: handle blockedSystemAddresses mapping
-        // blockedSystemAddresses: [{key: '', value: ''}],
+        otherMessenger: ProxyL1CrossDomainMessenger.address
```
Comment (tynes) [inline packages/contracts-bedrock/tasks/genesis-l2.ts:251]: This file was deleted in a PR that was merged

**Comment by tynes** [comment]
I like this approach because it makes things cheaper and also resolves issues with storage slots since it moves a variable to being an immutable, we no longer need to worry about a storage slot collision.

@smartcontracts We should set the storage slot that is no longer being used to 0 as part of the L1 upgrade

**Comment by smartcontracts** [comment]
> @smartcontracts We should set the storage slot that is no longer being used to 0 as part of the L1 upgrade

`blockedSystemAddresses` was a new storage slot, so removing this would not require adding a spacer.

**Review by maurelian — APPROVED** [review]

**Comment by maurelian** [comment]
This looks safe to me. 
Regarding storage slot spacing etc, while these contracts are continuing to evolve I would propose that we continue pursuing the ideal design, and only worry about fixing any storage slot issues after the design is frozen.

**Pull request closed without merging**